src/utils/self_test/self_test.c

```diff
@@ -1635,141 +2986,563 @@ int parse_message_sizes_string(const char *pch,
+			ret = sscanf(optarg, "%d", &g_rep_count);
 			if (ret != 1) {
-				rep_count = default_rep_count;
+				g_rep_count = g_default_rep_count;
 				printf("Warning: Invalid repetitions-per-size\n"
 				       "  Using default value %d instead\n",
-				       rep_count);
+				       g_rep_count);
 			}
 			break;
 		case 'i':
-			ret = sscanf(optarg, "%d", &max_inflight);
+			ret = sscanf(optarg, "%d", &g_max_inflight);
 			if (ret != 1) {
-				max_inflight = default_max_inflight;
+				g_max_inflight = g_default_max_inflight;
 				printf("Warning: Invalid max-inflight-rpcs\n"
 				       "  Using default value %d instead\n",
-				       max_inflight);
+				       g_max_inflight);
 			}
 			break;
 		case 'a':
-			ret = sscanf(optarg, "%" SCNd16, &buf_alignment);
-			if (ret != 1 || buf_alignment < CRT_ST_BUF_ALIGN_MIN ||
-			    buf_alignment > CRT_ST_BUF_ALIGN_MAX) {
+			ret = sscanf(optarg, "%" SCNd16, &g_buf_alignment);
+			if (ret != 1 ||
+			    g_buf_alignment < CRT_ST_BUF_ALIGN_MIN ||
+			    g_buf_alignment > CRT_ST_BUF_ALIGN_MAX) {
 				printf("Warning: Invalid align value %d;"
 				       " Expected value in range [%d:%d]\n",
-				       buf_alignment, CRT_ST_BUF_ALIGN_MIN,
+				       g_buf_alignment, CRT_ST_BUF_ALIGN_MIN,
 				       CRT_ST_BUF_ALIGN_MAX);
-				buf_alignment = CRT_ST_BUF_ALIGN_DEFAULT;
+				g_buf_alignment = CRT_ST_BUF_ALIGN_DEFAULT;
 			}
 			break;
 		case 'b':
-			output_megabits = 1;
-			break;
-		case 't':
+			g_output_megabits = 1;
 			break;
 		case 'p':
-			attach_info_path = optarg;
+			if (g_attach_info_path != NULL) {
+				/* Avoid checkpatch warning */
+				free(g_attach_info_path);
+			}
+			alloc_g_attach_info_path = false;
+			g_attach_info_path = optarg;
 			break;
 		case 'q':
 			g_randomize_endpoints = true;
 			break;
-		case 'n':
+		case 'v':  /* *** */
+			ret = sscanf(optarg, "%d", &g_raw_data);
+			g_raw_data = g_raw_data <
+				     SELF_TEST_MAX_RAW_DATA_OUTPUT ?
+				     g_raw_data : SELF_TEST_MAX_RAW_DATA_OUTPUT;
+			break;
+		case 'w':  /* *** */
+			ret = sscanf(optarg, "%f", &g_scale_factor);
+			break;
+		case 'x':
+			if (alloc_g_expected_results) {
+				/* Avoid checkpatch warning */
+				free(g_expected_results);
+			}
+			alloc_g_expected_results = false;
+			g_expected_results = optarg;
+			break;
+		case 'y':
+			if (alloc_g_expected_infile) {
+				/* Avoid checkpatch warning */
+				free(g_expected_infile);
+			}
+			alloc_g_expected_infile = false;
+			g_expected_infile = optarg;
+			break;
+		case 'z':
+			if (alloc_g_expected_outfile) {
+				/* Avoid checkpatch warning */
+				free(g_expected_outfile);
+			}
+			alloc_g_expected_outfile = false;
+			g_expected_outfile = optarg;
+			break;
+		case 'o':
+			if (alloc_g_config_append) {
+				/* Avoid checkpatch warning */
+				free(g_config_append);
+			}
+			alloc_g_config_append = false;
+			g_config_append = optarg;
 			break;
+		case 'h':
 		case '?':
+#ifdef INCLUDE_OBSOLETE
+		case 'n':
+		case 't':
+#endif
+
+		default:
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment